tools/probes/probes_main.c

```diff
@@ -0,0 +1,306 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Adrian Bonislawski <adrian.bonislawski@intel.com>
+
+/*
+ * Probes will extract data for several probe points in one stream
+ * with extra headers. This app will read the resulting file,
+ * strip the headers and create wave files for each extracted buffer.
+ *
+ * Usage to parse data and create wave files: ./sof-probes -p data.bin
+ *
+ */
+
+#include <ipc/probe.h>
+#include <sof/math/numbers.h>
+#include "wave.h"
+
+#include <ctype.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#define APP_NAME "sof-probes"
+
+#define PACKET_MAX_SIZE	4096	/**< Size limit for probe data packet */
+#define DATA_READ_LIMIT 1024	/**< Data limit for file read */
+#define FILES_LIMIT	32	/**< Maximum num of files */
+
+struct wave_files {
+	FILE *fd;
+	uint32_t buffer_id;
+	uint32_t size;
+	struct wave header;
+};
+
+enum p_state {
+	READY = 0,		/**< At this stage app is looking for a SYNC word */
+	SYNC,			/**< SYNC received, copying data */
+	CHECK			/**< Check crc and save packet if valid */
+};
+
+static uint32_t sample_rate[] = {
+	8000, 11025, 12000, 16000, 22050, 24000, 32000, 44100,
+	48000, 64000, 88200, 96000, 128000, 176400, 192000
+};
+
+static void usage(void)
+{
+	fprintf(stdout, "Usage %s <option(s)> <buffer_id/file>\n\n", APP_NAME);
+	fprintf(stdout, "%s:\t -p file\tParse extracted file\n\n", APP_NAME);
+	fprintf(stdout, "%s:\t -h \t\tHelp, usage info\n", APP_NAME);
+	exit(0);
+}
+
+int write_data(char *path, char *data)
+{
+	FILE *fd;
+
+	fd = fopen(path, "w");
+	if (!fd) {
+		fprintf(stderr, "error: unable to open file %s, error %d\n",
+			path, errno);
+		return errno;
+	}
+
+	fprintf(fd, "%s", data);
+	fclose(fd);
+
+	return 0;
+}
+
+int get_buffer_file(struct wave_files *files, uint32_t buffer_id)
+{
+	int i;
+
+	for (i = 0; i < FILES_LIMIT; i++) {
+		if (files[i].buffer_id == buffer_id)
+			return i;
+	}
+	return -1;
+}
+
+int init_wave(struct wave_files *files, uint32_t buffer_id, uint32_t format)
+{
+	char path[64];
```

Comment:
This should be a macro and probably bigger

Reply:
done